Acceptance-Tests.md

```diff
@@ -55,34 +67,34 @@ Docker:
+2) Within the user directory, create a new file for each test. For ex. `create-new-topic.spec.ts` and `delete-topic.spec.ts` for `Topic Manager` user. And these top-level test contains single user stories checking their test steps and expectations mentioned in the [testing spreadsheet](https://docs.google.com/spreadsheets/d/1O8EHiSAGrG0yoNUBz9E4DIwKNS8Rfsv_ffC4k1WK5jc/edit?usp=sharing).
 
-3) The functionality of the top-level tests for each user-type is defined in the `user-utilities` directory. For ex. the blog editor tests are written within the `spec/blog-editor-tests` directory, and the functionality of the tests are defined in the `user-utilities/blog-post-editor-utils.js` file.
-> Note: A utility file is maintained for each user type. The purpose of maintaining this file is to add methods specific to that user on top of the already provided basic methods. This file maintains a user class which is extended from the base class of puppeteer-utils.js . For ex. blog-post-admin-utils.js have a class e2eBlogPostAdmin which have methods like `createDraftBlogPostWithTitle`, `deleteDraftBlogPostWithTitle` etc. specific to Blog Admin only.
+3) The functionality of the top-level tests for each user-type is defined in the `user-utilities` directory. For ex. the blog editor tests are written within the `spec/blog-editor-tests` directory, and the functionality of the tests are defined in the `user-utilities/blog-post-editor-utils.ts` file.
+> Note: A utility file is maintained for each user type. The purpose of maintaining this file is to add methods specific to that user on top of the already provided basic methods. This file maintains a user class which is extended from the base class of puppeteer-utils.ts . For ex. blog-post-admin-utils.ts have a class BlogPostEditor which have methods like `createDraftBlogPostWithTitle`, `deleteDraftBlogPostWithTitle` etc. specific to Blog Admin only.
 
 
-4) The utility files are imported in the top-level test files and the methods are called to perform the required actions. For ex. in the `create-draft-and-delete-draft-blog-post.spec.js` file, the `createDraftBlogPostWithTitle` method is called to create a draft blog post with a given title. The `deleteDraftBlogPostWithTitle` method is called to delete the draft blog post with the given title.
+4) The utility files are imported in the top-level test files and the methods are called to perform the required actions. For ex. in the `try-to-publish-a-duplicate-blog-post-and-get-blocked.spec.ts` file, the `createNewBlogPostWithTitle` method is called to create a new blog post with the given title. The `expectUserUnableToPublishBlogPost` method is called to check if the user is unable to publish a blog post.
 
```

Comment:
Maybe create a new section here to show how to add a new user-utilities file to the user-factory. First if the role requires a super admin to assign it, we should add the role in test-constants. Then, to add it, we would add the reference to the USER_ROLE_MAPPING (if the user requires a role from the super admin), otherwise if it is a role which doesn't like LoggedInUser, we should reference the developer to add the Factory to the array inside createNewUser under composeUserWithRoles(BaseUserFactory(), [...]).

Reply:
(5) Adding a New User-Utilities File to the User Factory:
If a new user-utilities file needs to be added to the user factory, follow these steps:

- Determine if the role requires a super admin to assign it. If it does, add the role to the `ROLES` object in the `test-constants.ts` file.

- Next, ensure the new utility class is referenced appropriately. If the user requires a role from the super admin, add the reference to the `USER_ROLE_MAPPING` object in `test-constants.ts`. If the role is not assigned by a super admin, but rather inherent to the user (like `LoggedInUser`), direct the developer to add the factory to the array inside `createNewUser` under `composeUserWithRoles(BaseUserFactory(), [...])`.

By following these steps, the new user-utilities file can be integrated into the user factory for use in tests.


@jnvtnguyen PTAL

Reply:
should be Roles in test-contants.ts and USER_ROLE_MAPPING inside the user-factory.ts.

Reply:
done